File: DSA_Questions/src/com/dsa_questions/Summer_internship.java
package com.dsa_questions;

import java.util.*;

public class Summer_internship {

	    public static String Summer(int[] arr){
	        
	        int five = 0;
	        
	        int ten = 0;
	        
	        for(int i = 0;i < arr.length;i++){
	            
	            if(arr[i] == 5){
	                
	                five++;
	                
	            }else if(arr[i] == 10){
	                
	                if(five >= 1){
	                    
	                five--;
	                
	                ten++;
	 
	                }else{
	                    
	                    return "NO";
	                    
	                }
	                
	            }else if(arr[i] == 20){
	                
	                if(ten >= 1 && five >= 1){
	                    
	                    ten--;
	                    
	                    five--;
	                    
	                }else if(five >= 3){
	                    
	                    five = five - 3;
	                    
	                }else{
	                    
	                    return "NO";
	                    
	                }
	                
	            }
	            
	            
	        }
	        
	        return "YES";
	        
	    }
	    
	    
	    public static void main(String[] args){
	        
	        Scanner sc = new Scanner(System.in);
	        
	        int tc = sc.nextInt();
	        
	        for(int i = 0; i < tc; i++){
	            
	            int n = sc.nextInt();
	            
	            int arr[] = new int[n];
	            
	            for(int j = 0;j < n ; j++){
	                
	                arr[j] = sc.nextInt();
	                
	            }
	            
	            String ans = Summer(arr);
	            
	            System.out.println(ans);
	            

	        }
	        
	        sc.close();
	        
	    }
	    
	}
	

File: DSA_Questions_Python/Summer_internship.py
def fun(arr):
    
    five = 0
    
    ten = 0
    
    for k in range(len(arr)):
        
        if arr[k] == 5:
            
            five = five + 1
            
        elif arr[k] == 10:
            
            if five >= 1:
                
                five = five - 1
                
                ten = ten + 1
                
            else:
                
                return "NO"
        
        elif arr[k] == 20:
            
            if ten >= 1 and five >= 1:
                
                ten = ten - 1
                
                five = five - 1
                
            elif five >= 3:
                
                five = five - 3
                
            else:
                
                return "NO"
                
    return "YES"                

tc = int(input())

for i in range(tc):
    
    n = int(input())
    
        
    arr = list(map(int,input().split()))

    a = fun(arr)
    
    print(a)